src/components/lib/Web3Status.js

```diff
@@ -1,46 +1,37 @@
-import React, { Component } from 'react'
-
-import { withWeb3, withAccount, withConnectDapp } from '../../wrappers/web3'
+import React, { Component, useReducer, useState } from 'react'
 import Check from '../svgs/Check'
+import { useWeb3React } from '@web3-react/core'
+import { ConnectWalletDialog } from './ConnectWalletDialog'
+
+export const Web3Status = (props) => {
+	const { active, error } = useWeb3React()
+
+	let [showConnectWallet, setShowConnectWallet] = useState(false)
 
-class Web3Status extends Component {
-  componentDidMount() {
-    this.props.connectDapp()
-  }
+	let body = <div>
+		<div className="web3-status loading">
+			Loading...
+		</div>
+	</div>
+	
+	if(!active) {
+		body = <div className="web3-status notify">
+			<span onClick={() => setShowConnectWallet(true)}>
+				Connect to a Wallet
+			</span>
+		</div>
+	}
 
-  render() {
-    const { account, loading, web3 } = this.props
+	else if(active) {
+		body = <div className="web3-status success">
+			<Check width="15px" /> Connected
+		</div>
+	}
 
-    if (loading) {
-      return (
-        <div className="web3-status loading">
-          loading...
-          </div>
-      )
-    }
-  
-    if (!web3) {
-      return (
-        <div className="web3-status alert">
-          Web3 not detected.  We suggest <a href="http://metamask.io" target="_blank" rel="noopener noreferrer">MetaMask</a>.
-          </div>
-      )
-    }
-  
-    if (!account) {
-      return (
-        <div className="web3-status notify">
-          Web3 detected, but you need to connect & log into an account.
-          </div>
-      )
-    }
-  
-    return (
-      <div className="web3-status success">
-        <Check width="15px" /> Account logged in
-      </div>
-    )
-  }
+	return <div>
+		<ConnectWalletDialog onConnected={() => setShowConnectWallet(false)} shown={showConnectWallet} />
```

Comment:
Something about this construction still feels a little weird, but I can't quite put my finger on it…

Reply:
Does it feel like Web3Status should not be managing the dialog visibility?

ie. `this.close` instead of `setShowConnectWallet`

[Semantic UI React](https://react.semantic-ui.com/modules/modal/#variations-dimmer) is an example of the former. It's certainly cleaner.

Reply:
That's a chunk of it, but I couldn't quite play that out to where I wanted it last night. Let's leave it as a niggle for later.